research-hub-web/src/app/components/layout/navbar/navbar.component.ts

```diff
@@ -0,0 +1,84 @@
+import { Component, EventEmitter, Input, OnDestroy, OnInit, Output } from '@angular/core';
+import { NavigationEnd, Router, RouterEvent } from '@angular/router';
+import { SearchBarService } from '@app/components/search-bar/search-bar.service';
+import { Category, Stage } from '@app/graphql/schema';
+import { HomeScrollService } from '@services/home-scroll.service';
+import { LoginService, UserInfoDto } from '@uoa/auth';
+import { from, Observable, Subscription } from 'rxjs';
+import { filter, switchMap, tap } from 'rxjs/operators';
+
+@Component({
+  selector: 'app-navbar',
+  templateUrl: './navbar.component.html',
+  styleUrls: ['./navbar.component.scss']
+})
+export class NavbarComponent implements OnInit, OnDestroy {
+  @Input() allCategories: Category[] = [];
+  @Input() allStages: Stage[] = [];
+
+  @Output() toggleSidenav: EventEmitter<void> = new EventEmitter<void>();
+
+  public isHome = false;
+  public currentUrl = '/';
+
+  public userInfo$: Observable<UserInfoDto>;
+  public loggedIn$: Observable<boolean>;
+
+  private subscriptions: Subscription = new Subscription();
+
+  constructor(
+    private router: Router,
+    public homeScrollService: HomeScrollService,
+    public searchBarService: SearchBarService,
+    public loginService: LoginService
+  ) { }
+
+  ngOnInit(): void {
+    this.subscriptions.add(
+      this.router.events.pipe(
+        filter((event: RouterEvent) => event instanceof NavigationEnd)
+      ).subscribe({
+        next: (event: NavigationEnd) => {
+          this.isHome = event.urlAfterRedirects ? (event.urlAfterRedirects === '/home') : false;
+          this.currentUrl = event.urlAfterRedirects;
+        }
+      })
+    );
+
+    this.loggedIn$ = from(this.loginService.isAuthenticated()).pipe(
+      switchMap(() => this.loginService.loggedIn$)
+    );
+
+    this.userInfo$ = this.loginService.userInfo$.pipe(
+      filter(userInfo => userInfo !== null && userInfo !== undefined),
+      tap(userInfo => this.sendGoogleAnalyticsUserInfo(userInfo))
+    );
+  }
+
+  private sendGoogleAnalyticsUserInfo(userInfo: UserInfoDto): void {
+    window.dataLayer.push({
+      'user': JSON.stringify(userInfo),
+      ...userInfo,
+    });
+
+    // groups disabled for now, as there is a bug in the UoA auth library
```

Comment:
I have a feeling this may break out current reporting on staff vs not staff. However that needs to be updated/modified anyway so I guess this ok for now? Any ETA on the UoA auth bug being fixed?

Reply:
@rosemcc Mohsin said it would take at least 3 weeks, which was sometime last week, I believe. So I hope it'll come out soon. What should we do about this for now? We could consider reworking the analytics so that we only track whether someone is signed in or not. I don't think we need to track individual groups?

Reply:
I think just leave as you've got it for now. We can deal with a few weeks of possibly not collecting accurate data.